### PR TITLE
Init using module template

### PR DESCRIPTION
**Pull request by jpuri**
Initialise using https://github.com/MetaMask/metamask-module-template/tree/main

### CODE REVIEW

**Review by mcmire — COMMENTED** [review]
Looks good! You may need to re-run `yarn install` to clear the CI failure though. I can approve after that.

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
+name: Publish Release
+
+on:
+  workflow_call:
+    secrets:
+      NPM_TOKEN:
```
[the remark this answers is not in the document]
Comment (jpuri) [reply .github/workflows/publish-release.yml:6]: @Gudahtt , @mcmire : I made you both admin of this repo.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
+{
+  "name": "@metamask/ppom-validator",
+  "version": "0.0.0",
+  "description": "The MetaMask Node module template",
```
[the remark this answers is not in the document]
Comment (jpuri) [reply package.json]: Done

**Comment by Gudahtt** [comment]
@SocketSecurity ignore jest@28.1.3
@SocketSecurity ignore jest-cli@28.1.3

The bin script confusion pointed out is just the same script being exposed in both places

**Comment by Gudahtt** [comment]
@SocketSecurity ignore @lavamoat/aa@3.1.2
@SocketSecurity ignore @lavamoat/allow-scripts@2.3.1
@SocketSecurity ignore @lavamoat/preinstall-always-fail@1.0.0
@SocketSecurity ignore @metamask/auto-changelog@3.1.0

These are our packages. They are expected to have postinstall scripts and usage of `fs`

**Comment by Gudahtt** [comment]
@SocketSecurity ignore @babel/parser@7.16.4
@SocketSecurity ignore @cspotcode/source-map-support@0.8.1
@SocketSecurity ignore @istanbuljs/load-nyc-config@1.1.0
@SocketSecurity ignore @nodelib/fs.scandir@2.1.5
@SocketSecurity ignore @nodelib/fs.stat@2.0.5
@SocketSecurity ignore @npmcli/fs@2.1.2
@SocketSecurity ignore @npmcli/move-file@2.0.1
@SocketSecurity ignore @npmcli/node-gyp@3.0.0
@SocketSecurity ignore @npmcli/run-script@6.0.2
@SocketSecurity ignore acorn@8.8.2
@SocketSecurity ignore ajv@6.12.6
@SocketSecurity ignore argparse@1.0.10
@SocketSecurity ignore argparse@2.0.1
@SocketSecurity ignore babel-plugin-istanbul@6.1.1
@SocketSecurity ignore bplist-parser@0.2.0
@SocketSecurity ignore bs-logger@0.2.6
@SocketSecurity ignore chokidar@3.5.3
@SocketSecurity ignore chownr@2.0.0
@SocketSecurity ignore commander@9.5.0
@SocketSecurity ignore convert-source-map@1.9.0
@SocketSecurity ignore cosmiconfig@7.1.0
@SocketSecurity ignore create-require@1.1.1
@SocketSecurity ignore cross-spawn@7.0.3
@SocketSecurity ignore default-browser-id@3.0.0
@SocketSecurity ignore depcheck@1.4.3
@SocketSecurity ignore escalade@3.1.1
@SocketSecurity ignore eslint@8.41.0
@SocketSecurity ignore eslint-module-utils@2.8.0
@SocketSecurity ignore eslint-plugin-import@2.27.5
@SocketSecurity ignore eslint-plugin-jest@27.2.1
@SocketSecurity ignore eslint-plugin-jsdoc@39.9.1
@SocketSecurity ignore eslint-plugin-node@11.1.0
@SocketSecurity ignore esprima@4.0.1
@SocketSecurity ignore fast-glob@3.2.12
@SocketSecurity ignore file-entry-cache@6.0.1
@SocketSecurity ignore flat-cache@3.0.4
@SocketSecurity ignore fs-minipass@2.1.0
@SocketSecurity ignore fs.realpath@1.0.0
@SocketSecurity ignore get-package-type@0.1.0
@SocketSecurity ignore glob@7.2.3
@SocketSecurity ignore glob@8.1.0
@SocketSecurity ignore infer-owner@1.0.4
@SocketSecurity ignore is-docker@2.2.1
@SocketSecurity ignore is-docker@3.0.0
@SocketSecurity ignore is-wsl@2.2.0
@SocketSecurity ignore isexe@2.0.0
@SocketSecurity ignore istanbul-lib-report@3.0.0
@SocketSecurity ignore istanbul-lib-source-maps@4.0.1
@SocketSecurity ignore istanbul-reports@3.1.5
@SocketSecurity ignore jest-haste-map@28.1.3

These all appear to be dev dependencies that have legitimate reasons to use `fs`. Can't see any red flags.

**Comment by Gudahtt** [comment]
@SocketSecurity ignore jest-it-up@2.2.0
    @SocketSecurity ignore js-yaml@4.1.0
    @SocketSecurity ignore js-yaml@3.14.1
    @SocketSecurity ignore json5@1.0.2
    @SocketSecurity ignore json5@2.2.3
    @SocketSecurity ignore locate-path@5.0.0
    @SocketSecurity ignore locate-path@6.0.0
    @SocketSecurity ignore loose-envify@1.4.0
    @SocketSecurity ignore lunr@2.3.9
    @SocketSecurity ignore make-dir@3.1.0
    @SocketSecurity ignore marked@4.3.0
    @SocketSecurity ignore mkdirp@1.0.4
    @SocketSecurity ignore node-gyp@9.3.1
    @SocketSecurity ignore path-exists@4.0.0
    @SocketSecurity ignore path-type@4.0.0
    @SocketSecurity ignore prettier@2.8.8
    @SocketSecurity ignore read-cmd-shim@4.0.0
    @SocketSecurity ignore readdirp@3.6.0
    @SocketSecurity ignore require-directory@2.1.1
    @SocketSecurity ignore resolve-from@4.0.0
    @SocketSecurity ignore resolve-from@5.0.0
    @SocketSecurity ignore rimraf@3.0.2
    @SocketSecurity ignore sass@1.62.1
    @SocketSecurity ignore shiki@0.14.2
    @SocketSecurity ignore source-map-support@0.5.13
    @SocketSecurity ignore tar@6.1.15
    @SocketSecurity ignore ts-jest@28.0.8
    @SocketSecurity ignore ts-node@10.9.1
    @SocketSecurity ignore tsconfig-paths@3.14.2
    @SocketSecurity ignore typedoc@0.23.28
    @SocketSecurity ignore typescript@4.8.4
    @SocketSecurity ignore v8-compile-cache-lib@3.0.1
    @SocketSecurity ignore v8-to-istanbul@9.1.0
    @SocketSecurity ignore walker@1.0.8
    @SocketSecurity ignore write-file-atomic@4.0.2
    @SocketSecurity ignore write-file-atomic@5.0.1
    @SocketSecurity ignore y18n@5.0.8
    @SocketSecurity ignore yargs@16.2.0
    @SocketSecurity ignore yargs@17.7.2
    @SocketSecurity ignore yargs-parser@20.2.9
    @SocketSecurity ignore yargs-parser@21.1.1
    @SocketSecurity ignore cacache@16.1.3
    @SocketSecurity ignore bin-links@4.0.1
    @SocketSecurity ignore cmd-shim@6.0.1
    @SocketSecurity ignore read-package-json-fast@3.0.2
    @SocketSecurity ignore http-proxy-agent@5.0.0
    @SocketSecurity ignore https-proxy-agent@5.0.1
    @SocketSecurity ignore socks-proxy-agent@7.0.0
    @SocketSecurity ignore make-fetch-happen@10.2.1
    @SocketSecurity ignore minipass-fetch@2.1.2

More expected fs access

**Comment by Gudahtt** [comment]
@SocketSecurity ignore-all

Well, I give up. It seems to come up with a new batch of innocuous warnings each time about expected activities of dev dependencies.

**Comment by jpuri** [comment]
Thanks a lot @Gudahtt , @mcmire 🙏

**Pull request merged**